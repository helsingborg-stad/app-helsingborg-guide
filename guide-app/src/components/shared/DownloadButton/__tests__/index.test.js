// @flow

/* eslint-disable import/first */
import "react-native";
import React from "react";
import { DownloadButton } from "../index";

import renderer from "react-test-renderer";

test("status => stopped", () => {
  const status: DownloadStatus = "stopped";
  const progress = 0;
  const tree = renderer
    .create(
      <DownloadButton
        status={status}
        progress={progress}
        startDownload={() => {}}
        resumeDownload={() => {}}
        cancelDownload={() => {}}
        pauseDownload={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("status => stopped, progress >= 1", () => {
  const status: DownloadStatus = "stopped";
  const progress = 2;
  const tree = renderer
    .create(
      <DownloadButton
        status={status}
        progress={progress}
        startDownload={() => {}}
        resumeDownload={() => {}}
        cancelDownload={() => {}}
        pauseDownload={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("status => pending, progress < 1", () => {
  const status: DownloadStatus = "pending";
  const progress = 0;
  const tree = renderer
    .create(
      <DownloadButton
        status={status}
        progress={progress}
        startDownload={() => {}}
        resumeDownload={() => {}}
        cancelDownload={() => {}}
        pauseDownload={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("status => pending, progress >= 1", () => {
  const status: DownloadStatus = "pending";
  const progress = 2;
  const tree = renderer
    .create(
      <DownloadButton
        status={status}
        progress={progress}
        startDownload={() => {}}
        resumeDownload={() => {}}
        cancelDownload={() => {}}
        pauseDownload={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("status => paused, progress < 1", () => {
  const progress = 0;
  const tree = renderer
    .create(
      <DownloadButton
        status="paused"
        progress={progress}
        startDownload={() => {}}
        resumeDownload={() => {}}
        cancelDownload={() => {}}
        pauseDownload={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("status => paused, progress >= 1", () => {
  const progress = 2;
  const tree = renderer
    .create(
      <DownloadButton
        status="paused"
        progress={progress}
        startDownload={() => {}}
        resumeDownload={() => {}}
        cancelDownload={() => {}}
        pauseDownload={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("status => done, progress < 1", () => {
  const progress = 0;
  const tree = renderer
    .create(
      <DownloadButton
        status="done"
        progress={progress}
        startDownload={() => {}}
        resumeDownload={() => {}}
        cancelDownload={() => {}}
        pauseDownload={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("status => done, progress >= 1", () => {
  const progress = 2;
  const tree = renderer
    .create(
      <DownloadButton
        status="done"
        progress={progress}
        startDownload={() => {}}
        resumeDownload={() => {}}
        cancelDownload={() => {}}
        pauseDownload={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
