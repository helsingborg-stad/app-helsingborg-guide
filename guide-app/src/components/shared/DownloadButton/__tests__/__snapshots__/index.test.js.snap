// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`status => done, progress < 1 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 2,
        "borderColor": "#D6D6D6",
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingBottom": "2%",
        "paddingHorizontal": "4%",
        "paddingTop": "2%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "green",
            "fontSize": 16,
          },
          undefined,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "green",
          "fontFamily": "Roboto",
          "fontSize": 16,
        }
      }
    >
      Available offline
    </Text>
  </View>
</View>
`;

exports[`status => done, progress >= 1 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 2,
        "borderColor": "#D6D6D6",
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingBottom": "2%",
        "paddingHorizontal": "4%",
        "paddingTop": "2%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "green",
            "fontSize": 16,
          },
          undefined,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "green",
          "fontFamily": "Roboto",
          "fontSize": 16,
        }
      }
    >
      Available offline
    </Text>
  </View>
</View>
`;

exports[`status => paused, progress < 1 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 2,
        "borderColor": "#D6D6D6",
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "paddingBottom": "3%",
        "paddingTop": "4%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#711A5B",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        Cancel
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#711A5B",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        Resume
      </Text>
    </View>
  </View>
  <View>
    <RCTProgressView
      progress={0}
      progressTintColor="#D35098"
      style={
        Array [
          Object {
            "height": 2,
          },
          Object {
            "transform": Array [
              Object {
                "scaleX": 1,
              },
              Object {
                "scaleY": 2.5,
              },
            ],
          },
        ]
      }
      trackTintColor="#00000000"
    />
  </View>
</View>
`;

exports[`status => paused, progress >= 1 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 2,
        "borderColor": "#D6D6D6",
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "paddingBottom": "3%",
        "paddingTop": "4%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#711A5B",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        Cancel
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#711A5B",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        Resume
      </Text>
    </View>
  </View>
  <View>
    <RCTProgressView
      progress={2}
      progressTintColor="#D35098"
      style={
        Array [
          Object {
            "height": 2,
          },
          Object {
            "transform": Array [
              Object {
                "scaleX": 1,
              },
              Object {
                "scaleY": 2.5,
              },
            ],
          },
        ]
      }
      trackTintColor="#00000000"
    />
  </View>
</View>
`;

exports[`status => pending, progress < 1 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 2,
        "borderColor": "#D6D6D6",
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingBottom": "2%",
        "paddingHorizontal": "4%",
        "paddingTop": "4%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#711A5B",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#711A5B",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "500",
            "lineHeight": 23,
            "marginRight": 13,
          }
        }
      >
        Downloading 0% (touch to pause)
      </Text>
    </View>
  </View>
  <View>
    <RCTProgressView
      progress={0}
      progressTintColor="#D35098"
      style={
        Array [
          Object {
            "height": 2,
          },
          Object {
            "transform": Array [
              Object {
                "scaleX": 1,
              },
              Object {
                "scaleY": 2.5,
              },
            ],
          },
        ]
      }
      trackTintColor="#00000000"
    />
  </View>
</View>
`;

exports[`status => pending, progress >= 1 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 2,
        "borderColor": "#D6D6D6",
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingBottom": "2%",
        "paddingHorizontal": "4%",
        "paddingTop": "4%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#711A5B",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#711A5B",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "500",
            "lineHeight": 23,
            "marginRight": 13,
          }
        }
      >
        Downloading 100% (touch to pause)
      </Text>
    </View>
  </View>
  <View>
    <RCTProgressView
      progress={2}
      progressTintColor="#D35098"
      style={
        Array [
          Object {
            "height": 2,
          },
          Object {
            "transform": Array [
              Object {
                "scaleX": 1,
              },
              Object {
                "scaleY": 2.5,
              },
            ],
          },
        ]
      }
      trackTintColor="#00000000"
    />
  </View>
</View>
`;

exports[`status => stopped 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 2,
        "borderColor": "#D6D6D6",
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingBottom": "2%",
        "paddingHorizontal": "4%",
        "paddingTop": "4%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#711A5B",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#711A5B",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "500",
            "lineHeight": 23,
            "marginRight": 13,
          }
        }
      >
        Download
      </Text>
    </View>
  </View>
  <View>
    <RCTProgressView
      progress={0}
      progressTintColor="#D35098"
      style={
        Array [
          Object {
            "height": 2,
          },
          Object {
            "transform": Array [
              Object {
                "scaleX": 1,
              },
              Object {
                "scaleY": 2.5,
              },
            ],
          },
        ]
      }
      trackTintColor="#00000000"
    />
  </View>
</View>
`;

exports[`status => stopped, progress >= 1 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 2,
        "borderColor": "#D6D6D6",
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingBottom": "2%",
        "paddingHorizontal": "4%",
        "paddingTop": "4%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#711A5B",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#711A5B",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "500",
            "lineHeight": 23,
            "marginRight": 13,
          }
        }
      >
        Download
      </Text>
    </View>
  </View>
  <View>
    <RCTProgressView
      progress={2}
      progressTintColor="#D35098"
      style={
        Array [
          Object {
            "height": 2,
          },
          Object {
            "transform": Array [
              Object {
                "scaleX": 1,
              },
              Object {
                "scaleY": 2.5,
              },
            ],
          },
        ]
      }
      trackTintColor="#00000000"
    />
  </View>
</View>
`;
